### ISSUE TITLE
Why does it take so long to build?

### ISSUE THREAD

**Issue by Mradr** [dinosec/iphone-dataprotection#23]
Another issue I am seeing is that it is taking a hell of a long time to generate some type of key on start? I don't remember this being here before or taking that long from last time I look at the project, but whatever change - might really look into getting it faster using C or fixing why it is taking so long to do so.

**Comment by Mradr** [comment]
Looking more into the issues it does seem to be with PBKDF2 taking a large amount of time... I found this https://github.com/Ayrx/python-fastpbkdf2 but unable to build it right now to test to see if that would help improve the speed at all. Was wondering if someone could build and test to see if it does. 

![backup_tool](https://user-images.githubusercontent.com/937790/31546755-49f814da-afea-11e7-9893-1d00e41b87c3.png)

**Comment by Commodore1024** [comment]
You are correct. This implementation of PBKDF2 is especially slow. Pycryptodome 3.5 also features a faster implementation, per the release notes.
